aten/src/ATen/native/Distance.cpp

```diff
@@ -189,8 +189,8 @@ Tensor _cdist_backward(const Tensor& grad, const Tensor& _x1, const Tensor& _x2,
     x2 = x2.expand(tensor2_expand_size).contiguous();
   }
 
-  TORCH_CHECK(x1.is_contiguous(), "_cdist_backward requires X1 to be contiguous");
-  TORCH_CHECK(x2.is_contiguous(), "_cdist_backward requires X2 to be contiguous");
+  x1 = x1.contiguous();
+  x2 = x2.contiguous();
```

Comment:
Should the two below also be contiguous?

Reply:
And also, we can remove the `contiguous` calls in the lines just before this one.

Reply:
They have to be contiguous but I'm not sure if they could be non-contiguous...
`cdist` is the result of forward pass and IMO is always contiguous. I'm not sure about `grad`...
We can make them similarly as `cdist = cdist.contiguous()` and `grad = grad.contiguous()` which will be a no-op in most of the time (i hope)

Reply:
Yes, makes sense, Mario.

Reply:
grad can be non-contiguous for sure. `cdist(...).sum().backward()` is a good test to check that case.

Reply:
@albanD sorry, i do not quite get your conclusion. As `cdist(...).sum().backward()` on contig and now on non-contig inputs does not fail it means that `grad` was contiguous, right ? But I agree that we can get a non-contig `grad` from previous layer.
So, would we like to make 
```
cdist = cdist.contiguous()
grad = grad.contiguous()
```
instead of torch checks ?

Reply:
> instead of torch checks ?

Yes

> on contig and now on non-contig inputs does not fail it means that grad was contiguous, right ?

sum definitely generates non-contiguous gradients. Maybe it does call contiguous somewhere else.